### PR TITLE
Integration test for Search-flight endpoint

### PR DESCRIPTION
**Pull request by Alvtag**

### CODE REVIEW

File: traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java [hunk 1]
```diff
@@ -18,17 +14,14 @@
...
 
-import java.io.OutputStream;
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.util.*;
 
-import static com.guestlogix.travelercorekit.utilities.UrlHelper.urlEncodeUTF8;
-
 
 public class Router {
```
Comment (Alvtag) [inline traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java]: I changed Router.java to use a private getSessionToken that handles the edge case of a session's Token being null. 
I don't think this will happen in the app, since the lib is initialized early on; but in the instrumented test the API call happens very quickly, before the token is fetched. 

`.build()` with a null token value seems to be fine; the task naturally fetches a new token.

File: traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java [hunk 2]
```diff
@@ -382,4 +373,12 @@ private static Error transformError(Error error) {
             return error;
         }
     }
+
+    private static String getSessionToken(Session session) {
+        Token token = session.getToken();
+        if (null == token) {
+            Log.w(TAG, "Session token is null");
```
Comment (Alvtag) [inline traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java]: I'm using android.util.Log to be consistent with the rest of the class; but should we use TravelerLog.java throughout this class?
